config/src/main/java/com/typesafe/config/impl/ConfigDocumentParser.java

```diff
@@ -311,6 +316,47 @@ private boolean isKeyValueSeparatorToken(Token t) {
             }
         }
 
+        private ConfigNodeConditional parseConditional(ArrayList<AbstractConfigNode> children) {
+            Token openToken = nextTokenCollectingWhitespace(children);
+            if (openToken != Tokens.OPEN_SQUARE)
+                throw parseError("expecting [ after if");
+            children.add(new ConfigNodeSingleToken(openToken));
+
+            Token subToken = nextTokenCollectingWhitespace(children);
+            if (!Tokens.isSubstitution(subToken))
+                throw parseError("left side of conditional expression must be a variable substitution");
+            children.add(new ConfigNodeSingleToken(subToken));
+
+            Token eqToken = nextTokenCollectingWhitespace(children);
+            Token eq2Token = nextTokenCollectingWhitespace(children);
+            if (eqToken != Tokens.EQUALS || eq2Token != Tokens.EQUALS)
+                throw parseError("conditional check must be `==`");
+            children.add(new ConfigNodeSingleToken(eqToken));
+            children.add(new ConfigNodeSingleToken(eq2Token));
+
+            Token valToken = nextTokenCollectingWhitespace(children);
+            if (!Tokens.isValueWithType(valToken, ConfigValueType.STRING)
+                    && !Tokens.isUnquotedText(valToken)
+                    && !Tokens.isValueWithType(valToken, ConfigValueType.BOOLEAN))
+                throw parseError("right side of conditional expression must be a string or boolean");
+            children.add(new ConfigNodeSimpleValue(valToken));
+
+            Token closeToken = nextTokenCollectingWhitespace(children);
+            if (closeToken != Tokens.CLOSE_SQUARE)
+                throw parseError("expecting ] after conditional expression");
+            children.add(new ConfigNodeSingleToken(closeToken));
+
+            Token openCurlyToken = nextTokenCollectingWhitespace(children);
+            if (openCurlyToken != Tokens.OPEN_CURLY)
+                throw parseError("must open a conditional body using {");
```

Comment:
That's handled via the parseObject call on line 355.  The "true" argument specifies that we've already parsed the opening {.